File: back/src/models/LessonPreparation.js
import mongoose from 'mongoose';

const lessonPreparationSchema = new mongoose.Schema(
  {
    id: { type: String, index: true },
    lessonName: { type: String, required: true },
    objectives: { type: String, default: '' },
    deadline: { type: String, default: '' },
    serviceName: { type: String, default: 'all', index: true },
    className: { type: String, default: 'all', index: true },
    stageName: { type: String, default: 'all', index: true },
    serviceYear: { type: String, default: '2026', index: true },
    createdAt: { type: String, default: () => new Date().toISOString() },
  },
  {
    timestamps: true,
    strict: false,
    collection: 'preparations',
  }
);

lessonPreparationSchema.index({ serviceYear: 1, serviceName: 1 });

const LessonPreparation = mongoose.models.LessonPreparation || mongoose.model('LessonPreparation', lessonPreparationSchema);
export default LessonPreparation;

File: back/Models/LessonPreparation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("preparations")]
    public class LessonPreparation
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("lessonName")]
        public string LessonName { get; set; } = string.Empty;

        [Column("objectives")]
        public string Objectives { get; set; } = string.Empty;

        [Column("deadline")]
        public string Deadline { get; set; } = string.Empty;

        [Column("serviceName")]
        public string ServiceName { get; set; } = "all";

        [Column("className")]
        public string ClassName { get; set; } = "all";

        [Column("stageName")]
        public string StageName { get; set; } = "all";

        [Column("serviceYear")]
        public string ServiceYear { get; set; } = "2026";

        [Column("createdAt")]
        public string CreatedAt { get; set; } = DateTime.UtcNow.ToString("o");

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
